ww3tools/modelBuoy_collocation.py

```diff
@@ -87,748 +108,941 @@
+                else:
+                    mdm = np.copy(mhs) * np.nan
+
+            else:
+
+                if mhs.shape[1] == result['Hs'].shape[0]:
+                    stname = np.append(stname, np.atleast_1d(np.array(result['station_name'])))
+                    mhs = np.append(mhs, [result['Hs']], axis=0)
+                    mtp = np.append(mtp, [result['Tp']], axis=0)
+
+                    if 'wind_spd' in result.keys():
+                        mwn = np.append(mwn, [result['wind_spd']], axis=0)
+                    else:
+                        mwn = np.append(mwn, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                    if 'wind_dir' in result.keys():
+                        mwd = np.append(mwd, [result['wind_dir']], axis=0)
+                    else:
+                        mwd = np.append(mwd, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                    if 'freq' in result.keys():
+                        mfreq = np.append(mfreq, [result['freq']], axis=0)
+                    else:
+                        mfreq = np.append(mfreq, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                    if 'tm' in result.keys():
+                        mtm = np.append(mtm, [result['tm']], axis=0)
+                    else:
+                        mtm = np.append(mtm, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                    if 'dp' in result.keys():
+                        mdp = np.append(mdp, [result['dp']], axis=0)
+                    else:
+                        mdp = np.append(mdp, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                    if 'dm' in result.keys():
+                        mdm = np.append(mdp, [result['dm']], axis=0)
+                    else:
+                        mdm = np.append(mdm, [np.copy(result['Hs']) * np.nan], axis=0)
+
+                else:
+                    print("   Stations in " + wlist[i] + " do not match the other tar files. Skipped " + wlist[i])
+
+
+    else:
+        gridinfo = int(0)
+        cyclonemap = int(0)
+        wlist = []
+        ftag = ''
+        forecastds = 0
+
+        if len(sys.argv) < 3:
+            sys.exit('At least one argument (list of ww3 files) must be informed.')
+        wlist = np.atleast_1d(np.loadtxt(sys.argv[1], dtype=str))
+        ftag = str(sys.argv[1]).split('list')[1].split('.txt')[0]
+        print('Reading ww3 list ' + str(sys.argv[1]))
+        print('Tag ' + ftag)
+
+        forecastds = int(sys.argv[2]) if len(sys.argv) >= 4 else 0
+        if forecastds > 0:
+            print('Forecast-type data structure')
+
+        gridinfo = str(sys.argv[3]) if len(sys.argv) >= 5 else int(0)
+        if gridinfo != 0:
+            print('Writing gridinfo ' + gridinfo)
+
+        cyclonemap = str(sys.argv[4]) if len(sys.argv) >= 6 else int(0)
+        if cyclonemap != 0:
+            print('Writing cyclonemap ' + cyclonemap)
+
+        # Paths
+        ndbcp = str(sys.argv[-1])  # Always get the last argument for ndbcp path
```

Comment:
@Ghazal-Mohammadpour This part seems to change the previous instructions for collocating buoy with bull_tar format, is that right?